src/modules/azure.rs

```diff
@@ -0,0 +1,301 @@
+use encoding_rs::*;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::path::PathBuf;
+
+use super::{Context, Module, RootModuleConfig};
+
+type JValue = serde_json::Value;
+
+use crate::configs::azure::AzureConfig;
+use crate::formatter::StringFormatter;
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("azure");
+    let config = AzureConfig::try_load(module.config);
+
+    let subscription_name = get_azure_subscription_name(context)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "subscription" => Some(Ok(subscription_name.to_string())),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `azure`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_azure_config_file_location(context: &Context) -> Option<PathBuf> {
+    context
+        .get_env("AZURE_CONFIG_DIR")
+        .map(PathBuf::from)
+        .or_else(|| {
+            let mut home = dirs_next::home_dir()?;
+            home.push(".azure");
+            Some(home)
+        })
+}
+
+fn get_azure_subscription_name(context: &Context) -> Option<String> {
+    let mut azure_profile_json = get_azure_config_file_location(context)?;
+    azure_profile_json.push("azureProfile.json");
+
+    let mut cloud_config = get_azure_config_file_location(context)?;
+    cloud_config.push("clouds.config");
+
+    let file = File::open(&cloud_config).ok()?;
+    let reader = BufReader::new(file);
+    let cloud_config_lines = reader.lines().filter_map(Result::ok);
+
+    let subscription_line = cloud_config_lines
+        .skip_while(|line| line != "[AzureCloud]")
+        .find(|line| line.starts_with("subscription"))?;
+
+    let current_subscription_id = subscription_line.split('=').nth(1)?.trim();
+
+    let file = File::open(&azure_profile_json).ok()?;
+    let mut buffer: Vec<u8> = Vec::new();
+    let mut reader = BufReader::new(file);
+    reader.read_to_end(&mut buffer).ok()?;
+    let bytes = buffer.as_mut_slice();
+    let decodedbuffer = UTF_8.decode_with_bom_removal(bytes).0;
```

Comment:
Instead of adding a dependency, please use something like `.strip_prefix([239, 187, 191])` to get rid of the BOM.